octopoes/nibbles/definitions.py

```diff
@@ -0,0 +1,95 @@
+import importlib
+import pkgutil
+from logging import getLogger
+from pathlib import Path
+from types import MethodType, ModuleType
+
+from pydantic import BaseModel
+
+from octopoes.models import OOI
+
+NIBBLES_DIR = Path(__file__).parent
+NIBBLE_ATTR_NAME = "NIBBLE"
+NIBBLE_FUNC_NAME = "nibble"
+logger = getLogger(__name__)
+
+
+class NibbleParameterDefinition(BaseModel):
+    ooi_type: type[OOI]
+    relation_path: str | None = None
+
+    def __eq__(self, other):
+        if isinstance(other, NibbleParameterDefinition):
+            return vars(self) == vars(other)
+        elif isinstance(other, type):
+            return self.ooi_type == other
+        else:
+            return False
+
+    def __hash__(self):
+        return hash(str(self.ooi_type) + self.relation_path if self.relation_path else "\0")
+
+
+class NibbleDefinition:
```

Comment:
I was wondering, what is the reason this isn't implemented as an e.g. Pydantic class but instead as a POJO-like class?

Reply:
Somehow the Pydantic class does not work well with the importlib yielding the payload... not sure why but it fixed the issues so I moved on -- perhaps hoping one day you would fix it ;)